web_ui/src/components/UsageBillingPage.tsx

```diff
@@ -749,6 +753,210 @@ function SubscriptionUpsellPrompt({
+    e.preventDefault()
+    updateStripeCustomerInfo({ email: billingEmail })
+  }
+  return (
+    <Card className={className}>
+      <Card.Body>
+        <Card.Title>Billing Email</Card.Title>
+        <Form onSubmit={handleSubmit}>
+          <Form.Group>
+            <Form.Control
+              type="email"
+              required
+              value={billingEmail}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setBillingEmail(e.target.value)
+              }
+            />
+            <Form.Text className="text-muted">
+              Required. Address to send billing receipts.
+            </Form.Text>
+          </Form.Group>
+          <KodiakSaveButton state={apiState} />
+        </Form>
+      </Card.Body>
+    </Card>
+  )
+}
+
+function CompanyNameForm({
+  defaultValue,
+  className,
+}: {
+  readonly defaultValue: string
+  readonly className?: string
+}) {
+  const [companyName, setCompanyName] = React.useState(defaultValue)
+  const [apiState, updateStripeCustomerInfo] = useTeamApiMutation(
+    Current.api.updateStripeCustomerInfo,
+  )
+  function handleSubmit(e: React.ChangeEvent<HTMLFormElement>) {
+    e.preventDefault()
+    updateStripeCustomerInfo({ name: companyName })
+  }
+  return (
+    <Card className={className}>
+      <Card.Body>
+        <Card.Title>Company Name</Card.Title>
+        <Form onSubmit={handleSubmit}>
+          <Form.Group>
+            <Form.Control
+              type="text"
+              value={companyName}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setCompanyName(e.target.value)
+              }
+            />
+            <Form.Text className="text-muted">
+              Added to billing receipts if provided.
+            </Form.Text>
+          </Form.Group>
+          <KodiakSaveButton state={apiState} />
+        </Form>
+      </Card.Body>
+    </Card>
+  )
+}
+
+function PostalAddressForm({
+  defaultValue = {},
+  className,
+}: {
+  readonly defaultValue: {
+    readonly line1?: string
+    readonly line2?: string
+    readonly city?: string
+    readonly state?: string
+    readonly postalCode?: string
+    readonly country?: string
+  }
+  readonly className?: string
+}) {
+  const [line1, setLine1] = React.useState(defaultValue.line1 ?? "")
+  const [line2, setLine2] = React.useState(defaultValue.line2 ?? "")
+  const [city, setCity] = React.useState(defaultValue.city ?? "")
+  const [state, setState] = React.useState(defaultValue.state ?? "")
+  const [postalCode, setPostalCode] = React.useState(
+    defaultValue.postalCode ?? "",
+  )
+  const [country, setCountry] = React.useState(defaultValue.country ?? "")
+
+  const [apiState, updateStripeCustomerInfo] = useTeamApiMutation(
+    Current.api.updateStripeCustomerInfo,
+  )
+  function handleSubmit(e: React.ChangeEvent<HTMLFormElement>) {
+    e.preventDefault()
+    updateStripeCustomerInfo({
```

Comment:
how does this handle api failure?

Shouldn't it undo the changes when there is an error?

Reply:
We display error text "save failed" via the button component.

Reply:
<img width="371" alt="image" src="https://user-images.githubusercontent.com/1929960/84602686-3bfc4e00-ae57-11ea-9c0a-dc61b88cb0cc.png">